awswrangler/postgresql.py

```diff
@@ -347,13 +361,18 @@ def to_sql(
                 df.reset_index(level=df.index.names, inplace=True)
             column_placeholders: str = ", ".join(["%s"] * len(df.columns))
             insertion_columns = ""
+            upsert_str = ""
             if use_column_names:
                 insertion_columns = f"({', '.join(df.columns)})"
+            if mode == "upsert":
+                upsert_columns = ", ".join(df.columns.map(lambda column: f"{column}=EXCLUDED.{column}"))
+                conflict_columns = ", ".join(upsert_conflict_columns)  # type: ignore
```

Comment:
This should fail if None... `ON CONFLICT` clause is optional according to the [docs](https://www.postgresql.org/docs/devel/sql-insert.html) - so do you think there is value in supporting upsert w/o making conflict resolution mandatory?

Reply:
If `upsert_conflict_columns` is `None`, this could would not be reached because of L364, which I think is a reasonable behavior as without the `ON CONFLICT` it would just be a default insert. But maybe a test could be added that the Exception is raised correctly.

Reply:
Cool, I've approved the changes - feel free to add a test case when it's convenient.

Reply:
Added to the test case - now merging